ludwig/datasets/loaders/av_mnist.py

```diff
@@ -0,0 +1,214 @@
+                        self.data = np.load(os.path.join(self.raw_dataset_dir, self.modal, "train_data.npy"))
+                        self.labels = np.load(os.path.join(self.raw_dataset_dir, "train_labels.npy"))
+                    else:
+                        self.data = np.load(os.path.join(self.raw_dataset_dir, self.modal, "test_data.npy"))
+                        self.labels = np.load(os.path.join(self.raw_dataset_dir, "test_labels.npy"))
+
+                    if self.modal == "audio":
+                        self.data = self.data[:, np.newaxis, :, :]
+                    elif self.modal == "image":
+                        self.data = self.data.reshape(self.data.shape[0], 1, 28, 28)
+                    self.write_output_dataset(self.labels, self.data, os.path.join(self.raw_dataset_dir, dataset))
+                else:
+                    raise ValueError("the value of modal should be given")
+
+        return super().transform_files(file_paths)
+
+    def load_unprocessed_dataframe(self, file_paths: List[str]) -> pd.DataFrame:
+        """Load dataset files into a dataframe."""
+        return self.output_training_and_test_data()
+
+    def process_source_dataset(self, raw_dataset_dir="."):
+        """An intermediate method to write .npy files for the training and the test image and label data to the raw dataset directory
+        :args:
+            raw_dataset_dir (str) : the directory where all the raw data lives
+        :returns:
+            None"""
+        file_names = {
+            "train_data": "train-images-idx3-ubyte.gz",
+            "train_labels": "train-labels-idx1-ubyte.gz",
+            "test_data": "t10k-images-idx3-ubyte.gz",
+            "test_labels": "t10k-labels-idx1-ubyte.gz",
+        }
+        print("Raw dataset working directory: %s" % raw_dataset_dir)
+
+        for key, file_name in self.file_names.items():
+            file_path = os.path.join(raw_dataset_dir, file_name)
+            print("file: %s" % key)
+            f = open(file_path)
+            # read the definition of idx1-ubyte and idx3-ubyte
+            f.seek(4)
+            num = f.read(4)
+            num = int().from_bytes(num, "big")
+            print("size of %s : %d" % (key, num))
+            if re.match(r".*data.*", key) is not None:
+                height = f.read(4)
+                height = int().from_bytes(height, "big")
+                width = f.read(4)
+                width = int().from_bytes(width, "big")
+
+                data = np.frombuffer(f.read(), np.uint8).reshape(num, height, width)
+
+                # PCA projecting with 75% energy removing
+                n_comp = int(height * width)
+                pca = PCA(n_components=n_comp)
+                projected = pca.fit_transform(data.reshape(num, height * width))
+                n_comp = ((np.cumsum(pca.explained_variance_ratio_) > 0.25) != 0).argmax()
+                rec = np.matmul(projected[:, :n_comp], pca.components_[:n_comp])
```

Comment:
I will dig into this a bit more and let you know my thoughts on path forward

Reply:
Hey @skanjila, I had a chance to chat with @justinxzhao and @dantreiman and we collectively felt that it's not worth keeping PCA in the dataloader module. Here's some reasons why:

1. It seems like this preprocessing step for MNIST is from [one paper](https://github.com/slyviacassell/_MFAS) only, so it feels very specific to that particular paper.
2. It's not clear whether this preprocessing strategy is state-of-the-art
3. Doing PCA to only keep the top 75% of pixels that explain the image is a preprocessing step, so it shouldn't be part of the data loading step. One thing we could do is add it within Ludwig as a preprocessing step that users can do downstream, independent of the data loader itself. 
4. As a Ludwig user, you can do the same PCA step after loading the dataset manually (when using Ludwig in a programmatic way), and it might be better to leave this choice to the end user. 

What do you think?